pinecone/index_connection_test.go

```diff
@@ -555,3 +557,244 @@ func TestMarshalDescribeIndexStatsResponse(t *testing.T) {
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := toVector(tt.vector)
+			assert.Equal(t, tt.expected, result, "Expected result to be '%s', but got '%s'", tt.expected, result)
+		})
+	}
+}
+
+func TestToSparseValuesUnit(t *testing.T) {
+	tests := []struct {
+		name         string
+		sparseValues *data.SparseValues
+		expected     *SparseValues
+	}{
+		{
+			name:         "Pass nil sparse values, expect nil to be returned",
+			sparseValues: nil,
+			expected:     nil,
+		},
+		{
+			name: "Pass sparse values",
+			sparseValues: &data.SparseValues{
+				Indices: []uint32{0, 2},
+				Values:  []float32{0.01, 0.03},
+			},
+			expected: &SparseValues{
+				Indices: []uint32{0, 2},
+				Values:  []float32{0.01, 0.03},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := toSparseValues(tt.sparseValues)
+			assert.Equal(t, tt.expected, result, "Expected result to be '%s', but got '%s'", tt.expected, result)
+		})
+	}
+}
+
+func TestToScoredVectorUnit(t *testing.T) {
+	tests := []struct {
+		name         string
+		scoredVector *data.ScoredVector
+		expected     *ScoredVector
+	}{
+		{
+			name:         "Pass nil scored vector, expect nil to be returned",
+			scoredVector: nil,
+			expected:     nil,
+		},
+		{
+			name: "Pass scored dense vector",
+			scoredVector: &data.ScoredVector{
+				Id:     "dense-1",
+				Values: []float32{0.01, 0.01, 0.01},
+				Score:  0.1,
+			},
+			expected: &ScoredVector{
+				Vector: &Vector{
+					Id:     "dense-1",
+					Values: []float32{0.01, 0.01, 0.01},
+				},
+				Score: 0.1,
+			},
+		},
+		{
+			name: "Pass scored sparse vector",
+			scoredVector: &data.ScoredVector{
+				Id: "sparse-1",
+				SparseValues: &data.SparseValues{
+					Indices: []uint32{0, 2},
+					Values:  []float32{0.01, 0.03},
+				},
+				Score: 0.2,
+			},
+			expected: &ScoredVector{
+				Vector: &Vector{
+					Id: "sparse-1",
+					SparseValues: &SparseValues{
+						Indices: []uint32{0, 2},
+						Values:  []float32{0.01, 0.03},
+					},
+				},
+				Score: 0.2,
+			},
+		},
+		{
+			name: "Pass scored hybrid vector",
+			scoredVector: &data.ScoredVector{
+				Id:     "hybrid-1",
+				Values: []float32{0.01, 0.02, 0.03},
+				SparseValues: &data.SparseValues{
+					Indices: []uint32{0, 2},
+					Values:  []float32{0.01, 0.03},
+				},
+				Score: 0.3,
+			},
+			expected: &ScoredVector{
+				Vector: &Vector{
+					Id:     "hybrid-1",
+					Values: []float32{0.01, 0.02, 0.03},
+					SparseValues: &SparseValues{
+						Indices: []uint32{0, 2},
+						Values:  []float32{0.01, 0.03},
+					},
+				},
+				Score: 0.3,
+			},
+		},
+		{
+			name: "Pass scored hybrid vector with metadata",
+			scoredVector: &data.ScoredVector{
+				Id:     "hybrid-metadata-1",
+				Values: []float32{0.01, 0.02, 0.03},
+				SparseValues: &data.SparseValues{
+					Indices: []uint32{0, 2},
+					Values:  []float32{0.01, 0.03},
+				},
+				Metadata: &structpb.Struct{
```

Comment:
@austin-denoble do you prefer this syntax of constructing the `metadata` obj better than how we do it in [the `toVector` test](https://github.com/pinecone-io/go-pinecone/pull/38/files#diff-6c29b62c08d897a1ab3deb830aea678bf3181775b462b536dc895a0dc3b9bd50R637)? Not sure which is more idiomatic in Go.

Reply:
I think either are probably fine, although for these I tend to like doing things inline better to co-locate things with the specific test variant. Up to you.

Reply:
yeah I think I like inline better too :)